src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpClientIntegration.cs

```diff
@@ -0,0 +1,768 @@
+        TargetMinimumVersion = Major4,
+        TargetMaximumVersion = Major4)]
+        public static object DeleteAsync(
+            object handler,
+            object uri,
+            object boxedCancellationToken,
+            int opCode,
+            int mdToken,
+            long moduleVersionPtr)
+        {
+            if (handler == null)
+            {
+                throw new ArgumentNullException(nameof(handler));
+            }
+
+            const string methodName = nameof(DeleteAsync);
+
+            var callOpCode = (OpCodeValue)opCode;
+            var httpClient = handler.GetInstrumentedType(HttpClientTarget);
+            var cancellationToken = (CancellationToken)boxedCancellationToken;
+
+            Func<HttpClient, Uri, CancellationToken, Task<HttpResponseMessage>> instrumentedMethod = null;
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<HttpClient, Uri, CancellationToken, Task<HttpResponseMessage>>>
+                       .Start(moduleVersionPtr, mdToken, opCode, methodName)
+                       .WithConcreteType(httpClient)
+                       .WithParameters(uri, cancellationToken)
+                       .WithNamespaceAndNameFilters(ClrNames.GenericTask, ClrNames.Uri, ClrNames.CancellationToken)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: HttpClientTarget,
+                    methodName: methodName,
+                    instanceType: handler.GetType().AssemblyQualifiedName);
+                throw;
+            }
+
+            return DeleteAsyncInternal(
+                instrumentedMethod,
+                reportedType: callOpCode == OpCodeValue.Call ? httpClient : handler.GetType(),
+                (HttpClient)handler,
+                (Uri)uri,
+                cancellationToken);
+        }
+
+        private static async Task<HttpResponseMessage> DeleteAsyncInternal(
+            Func<HttpClient, Uri, CancellationToken, Task<HttpResponseMessage>> deleteAsync,
+            Type reportedType,
+            HttpClient handler,
+            Uri uri,
+            CancellationToken cancellationToken)
+        {
+            using (var scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, "DELETE", uri, IntegrationName))
+            {
+                try
+                {
+                    if (scope != null)
+                    {
+                        // add distributed tracing headers to the HTTP request
+                        SpanContextPropagator.Instance.Inject(scope.Span.Context, handler.DefaultRequestHeaders);
```

Comment:
I think we have a blocking issue here. I'll ping the team so we can discuss.

Reply:
Short version:
- we should wrap the headers using the `Wrap()` extension method instead of adding an overload (see [my comment here](https://github.com/DataDog/dd-trace-dotnet/pull/711#discussion_r418210577))
- we should not inject the headers into `HttpClient.DefaultRequestHeaders` because that is not thread-safe and is shared with other requests

Reply:
Thanks for seeing that.